File: ConfidenceIntervals/CI_simulation.py
import numpy as np

# The number of Monte Carlo simulations
MC = 10000

k = 0
l = 0

for i in range(1, MC + 1):
    # Draw two random bubbles from a uniform distribution
    x = np.random.uniform(-5, 5, size=2)
    
    # Confidence interval based on the sampling distribution
    CI1 = np.mean(x) + np.array([-1, 1]) * (5 - 5 / np.sqrt(2))
    contained = (CI1[0] < 0) & (CI1[1] > 0)
    k += contained
    
    # Non-parametric confidence interval (distance between bubbles)
    CI2 = np.mean(x) + np.array([-1, 1]) * np.abs(np.diff(x)) / 2
    contained = (CI2[0] < 0) & (CI2[1] > 0)
    l += contained

print("Method 1:", f"{round(k/MC * 100, 1)}%")
print("Method 2:", f"{round(l/MC * 100, 1)}%")

File: ConfidenceIntervals/CI_simulation.R
# The number of Monte Carlo simulations
MC <- 10000

k <- 0
l <- 0
for(i in 1:MC){
  # Draw two random bubbles from a uniform distribution
  x <- runif(2, -5, 5)
  
  # Confidence interval based on the sampling distribution
  CI1 <- mean(x) + c(-1, 1) * (5 - 5 / sqrt(2))
  contained <- (CI1[1] < 0 & CI1[2] > 0)
  k <- k + contained
  
  # Non-parametric confidence interval (distance between bubbles)
  CI2 <- mean(x) + c(-1, 1) * abs(diff(x)) / 2
  contained <- (CI2[1] < 0 & CI2[2] > 0)
  l <- l + contained
}

cat("Method 1:", paste0(round(k/MC * 100, 1), "%\n"))
cat("Method 2:", paste0(round(l/MC * 100, 1), "%"))